elide-dbmanager/elide-dbmanager-multiplex/src/main/java/com/yahoo/elide/dbmanager/multiplex/MultiplexWriteTransaction.java

```diff
@@ -0,0 +1,174 @@
+import com.yahoo.elide.core.EntityDictionary;
+import com.yahoo.elide.core.exceptions.TransactionException;
+
+import com.google.common.collect.Lists;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.IdentityHashMap;
+import java.util.List;
+import java.util.Map.Entry;
+
+import javax.ws.rs.core.MultivaluedHashMap;
+
+/**
+ * Multiplex transaction handler.
+ */
+public class MultiplexWriteTransaction extends MultiplexTransaction {
+    private static final Object NEWLY_CREATED_OBJECT = new Object();
+    private final IdentityHashMap<Object, Object> clonedObjects = new IdentityHashMap<>();
+    private final MultivaluedHashMap<DatabaseManager, Object> dirtyObjects = new MultivaluedHashMap<>();
+
+    public MultiplexWriteTransaction(MultiplexManager multiplexManager) {
+        super(multiplexManager);
+    }
+
+    @Override
+    protected DatabaseTransaction beginTransaction(DatabaseManager databaseManager) {
+        // begin updating transaction
+        return databaseManager.beginTransaction();
+    }
+
+    @Override
+    public void save(Object entity) {
+        getTransaction(entity).save(entity);
+        dirtyObjects.add(this.multiplexManager.getSubManager(entity.getClass()), entity);
+    }
+
+    @Override
+    public void delete(Object entity) {
+        getTransaction(entity).delete(entity);
+        dirtyObjects.add(this.multiplexManager.getSubManager(entity.getClass()), entity);
+    }
+
+    @Override
+    public void commit() {
+        // flush all before commits
+        flush();
+
+        ArrayList<DatabaseManager> commitList = new ArrayList<>();
+        for (Entry<DatabaseManager, DatabaseTransaction> entry : transactions.entrySet()) {
+            try {
+                entry.getValue().commit();
+                commitList.add(entry.getKey());
+            } catch (Error | RuntimeException e) {
+                TransactionException transactionException = new TransactionException(e);
+                reverseTransactions(commitList, transactionException);
+                throw transactionException;
+            }
+        }
+        transactions.clear();
+    }
+
+    /**
+     * Reverse changes of last commit since not all transactions could commit
+     * @param restoreList List of database managers to reverse the last commit
+     * @param cause cause to add any suppressed exceptions
+     */
+    private void reverseTransactions(ArrayList<DatabaseManager> restoreList, Throwable cause) {
+        for (DatabaseManager databaseManager : restoreList) {
+            try (DatabaseTransaction tx = databaseManager.beginTransaction()) {
+                List<Object> list = dirtyObjects.get(databaseManager);
+                for (Object dirtyObject : list == null ? Collections.emptyList() : list) {
+                    Object cloned = clonedObjects.get(dirtyObject);
```

Comment:
This method should have copious amounts of comments. In particular, our Javadoc should explain our method (i.e. we try to "undo" what was done) and list caveats (i.e. we require proper `.equals()` implementation for this to be effective). 

Most importantly, users should be warned that this does not make any consistency guarantees and this should be considered "dangerous" unless you understand the ramifications of using such a thing. Considering the situation:

```text
A commits
B fails to commit
** Attempt to "undo" A commit **
A fails to commit
```

A and B will obviously be inconsistent at this point. Before using this, you need to understand the effects on your whole system in such a scenario.